Tests/NIOTests/BootstrapTest.swift

```diff
@@ -205,4 +205,38 @@ class BootstrapTest: XCTestCase {
         XCTAssertNoThrow(try childChannelDone.futureResult.wait())
         XCTAssertNoThrow(try serverChannelDone.futureResult.wait())
     }
+
+    func testClientTransportBootstrapAllowsConformanceCorrectly() throws {
```

Comment:
I was inspired by `ChannelTests.testBasicLifecycle`, happy to add more/different test cases if anyone has other ideas.